### PR TITLE
Death to UserAgents

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- Unblock consumers of AtlasDB that use `-Werror`
- Make more rapid progress on moving to a world with structured user-agents.

**Implementation Description (bullets)**:

**Testing (What was existing testing like?  What have you done to improve it?)**:

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**: yesterday

@leonz for SA

### CODE REVIEW

**Comment by jeremyk-91** [comment]
TODO: Docs updates, the docs still reference the old config.

**Comment by gmaretic** [comment]
Seems a bit draconian, punishment should fit the crime...

File: atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy [hunk 1]
```diff
@@ -217,7 +218,7 @@ class AtlasCoreModule implements AtlasConsoleModule {
     private setupConnection(AtlasDbConfig config) {
         TransactionManager tm = TransactionManagers.builder()
                 .config(config)
-                .userAgent("atlasdb console")
+                .structuredUserAgent(UserAgent.of(UserAgent.Agent.of("atlasdb console", "0.0.0")))
```
Comment (gmaretic) [inline atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy]: This is not a valid user agent, space not allowed

Comment (jeremyk-91) [reply atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy]: Someday, we'll have better Atlas Console tests

**Pull request merged**